src/server/pkg/work/work.go

```diff
@@ -363,9 +365,11 @@ func (w *Worker) subtaskFunc(subtaskKey string, processFunc ProcessFunc) subtask
 							}
 							subtaskInfo.Task = subtask
 							subtaskInfo.State = State_SUCCESS
+							subtaskInfo.Result = result
 							if retErr != nil {
 								subtaskInfo.State = State_FAILURE
 								subtaskInfo.Reason = retErr.Error()
+								subtaskInfo.Result = nil
```

Comment:
Lets not nil out the result here because there may be useful information that we want to collect when a failure occurs. Also, the process below should probably look like:
```
var err error
result, err = processFunc(claimCtx, subtask)
return err
```